src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/GetProfileNameDialog.xaml.cs

```diff
@@ -74,12 +74,14 @@ private void GetProfileNameDialogWindow_Loaded(object sender, RoutedEventArgs e)
             // We need to schedule this to occur later after databinding has completed, otherwise
             // focus appears in the textbox, but at the start of the suggested name rather than at
             // the end.
+#pragma warning disable VSTHRD001 // Avoid legacy threading switching APIs
```

Comment:
For a delegate like this where it's clear that no one waits on it and it doesn't have other dependencies, it's probably harmless to keep `Dispatcher.BeginInvoke` for this case. The analyzer suggests you avoid it in general, but there can be special cases (like a highly precise priority) that you might still want to use it.
That said, if you want a policy where you simply always use JTF.RunAsync instead (which I'm in favor of) one can write a JTF-derived type that sets this priority. We do this in VS already since it defines some custom priorities with `VsTaskRunContext`. It's trivial with an extension method in VS to get a JTF instance that uses any priority you specify. We just don't have one for `DispatcherPriority` yet. 
Filed https://github.com/Microsoft/vs-threading/issues/138